### PR TITLE
Add redux state 

### PR DESCRIPTION
**Pull request by carmenvramos**

### CODE REVIEW

File: src/store.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { createStore } from 'redux';
```
Comment (MariahAdams) [inline src/store.js]: Tests are required for reducers, actions and selectors.

**Comment by MariahAdams** [comment]
Container components should be separated from presentational components in their own folder.

File: src/components/markdown/Document.js [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
+  componentDidMount() {
+    this.unsubscribe = store.subscribe(() => {
+      this.setState({ markdown: getMarkdown(store.getState()) });
+    });
+  }
+
   updateMarkdown = ({ target }) => {
-    this.setState({ markdown: target.value });
+    store.dispatch(updateMarkdown(target.value));
   };
```
Comment (MariahAdams) [inline src/components/markdown/Document.js:27]: Missing component will unmount logic.

**Comment by MariahAdams** [comment]
Tests are required for reducers, actions and selectors.

File: src/components/markdown/Editor.js [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 import PropTypes from 'prop-types';
 import styles from './Editor.css';
+import { getMarkdown } from '../../selectors/document';
+import store from '../../store';
 
-function Editor({ markdown, updateMarkdown }) {
+function Editor({ updateMarkdown }) {
   return (
-    <textarea className={styles.Editor} value={markdown} onChange={updateMarkdown} />
+    <textarea className={styles.Editor} value={getMarkdown(store.getState())} onChange={updateMarkdown} />
```
Comment (MariahAdams) [inline src/components/markdown/Editor.js]: Functional components should not have state.

File: src/components/containers/Editor.js [hunk 4]
```diff
@@ -2,15 +2,15 @@ import React from 'react';
 import PropTypes from 'prop-types';
```
Comment (mehtaphysical) [inline src/components/containers/Editor.js]: this should be in `src/containers/Editor.js`

File: src/components/containers/Editor.js [hunk 5]
```diff
@@ -2,15 +2,15 @@ import React from 'react';
...
 import styles from './Editor.css';
 
-function Editor({ markdown, updateMarkdown }) {
+function Editor({ updateMarkdown, markdown }) {
```
Comment (mehtaphysical) [inline src/components/containers/Editor.js]: Functional components should not be containers. All containers are classes.

This one looks like a component

**Comment by MariahAdams** [comment]
Remember containers have state, presentational components do not. Please set up your file structure accordingly -- container folder should be on the same level as components.

File: src/components/containers/Preview.js [hunk 6]
```diff
@@ -1,15 +1,12 @@
 import React from 'react';
```
Comment (mehtaphysical) [inline src/components/containers/Preview.js]: This should be `src/containers/Preview.js`

File: src/components/containers/Preview.js [hunk 7]
```diff
@@ -1,15 +1,12 @@
...
-import PropTypes from 'prop-types';
 import marked from 'marked';
 import styles from 'github-markdown-css/github-markdown.css';
+import { getMarkdown } from '../../selectors/document';
+import store from '../../store';
 
-function Preview({ markdown }) {
-  const __html = marked(markdown);
+function Preview() {
+  const __html = marked(getMarkdown(store.getState()));
```
Comment (mehtaphysical) [inline src/components/containers/Preview.js]: This should be a class if it interacts with redux. It also needs to subscribe to state changes.

Does your app currently work in a browser?

**Pull request merged**